src/main/java/org/folio/linked/data/service/impl/MarcServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package org.folio.linked.data.service.impl;
+
+import java.util.Map;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.folio.linked.data.client.SrsClient;
+import org.folio.linked.data.service.MarcService;
+import org.folio.rest.jaxrs.model.Record;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class MarcServiceImpl implements MarcService {
+
+  private final SrsClient srsClient;
+
+  @Override
+  public Boolean isSupportedByInventoryId(String inventoryId) {
+    var response = srsClient.getSourceStorageRecordFormattedById(inventoryId);
+    return Optional.ofNullable(response.getBody())
+      .map(Record::getParsedRecord)
+      .map(parsedRecord -> (Map<?, ?>) parsedRecord.getContent())
+      .map(content -> (String) content.get("leader"))
+      .map(this::isMonograph)
+      .orElse(false);
+  }
+
+  private boolean isMonograph(String leader) {
+    return leader.charAt(6) == 'a' && (leader.charAt(7) == 'a' || leader.charAt(7) == 'm');
```

Comment:
minor: 
probably it's a bit more readable if reuse meaningful constants from:
RecordType.LANGUAGE_MATERIAL 
BibliographLevel.MONOGRAPHIC_COMPONENT_PART
BibliographLevel.MONOGRAPH_OR_ITEM
  
so this logic might look as following
```
private boolean isMonograph(String leader) {
    return isLanguageMaterial(leader.charAt(6)) 
      && isMonographicComponentPartOrItem(leader.charAt(7));
  }

  private boolean isLanguageMaterial(char typeOfRecord) {
    return typeOfRecord == RecordType.LANGUAGE_MATERIAL.value;
  }

  private boolean isMonographicComponentPartOrItem(char bibliographicLevel) {
    return bibliographicLevel == BibliographLevel.MONOGRAPHIC_COMPONENT_PART.value
      || bibliographicLevel == BibliographLevel.MONOGRAPH_OR_ITEM.value;
  }
```

Reply:
also with these changes we'll have several isMonograph() implementations, another one in marc4ld lib (org.folio.marc4ld.service.marc2ld.bib.MarcBib2LdMapperImpl), i think we can do refactoring later.

Reply:
done